app.rb

```diff
@@ -0,0 +1,136 @@
+require './person'
+require './teacher'
+require './student'
+require './rental'
+require './book'
+
+class App
+  attr_reader :books, :people
+
+  def initialize
+    @books = []
+    @people = []
+  end
+
+  def run
+    on_question_answer(on_questions)
+
+    run
+  end
+
+  def on_list_books
+    books.each do |book|
+      puts "Title: '#{book.title}', Author: '#{book.author}'"
+    end
+  end
+
+  def on_list_people
+    people.each do |person|
+      puts "[#{person.class.name}] ID: #{person.id} Name: #{person.name} Age: #{person.age}"
+    end
+  end
+
+  def add_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    create_person = gets.chomp
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+
+    add_person_general(create_person, age, name)
+
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] Please inspect the adding people logic in order to fix the raising error shown below. Look through how many arguments should be passed in order to make a successful person creation.
![main rb - OOPSchoolLibrary - Visual Studio Code 9_1_2023 2_55_32 PM](https://github.com/yemidada/OOPSchoolLibrary/assets/47625407/2410d0aa-aa32-4238-8984-8093c4ba2835)